packages/docker/src/generators/init/init.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  type Tree,
+  type GeneratorCallback,
+  readNxJson,
+  createProjectGraphAsync,
+  addDependenciesToPackageJson,
+  formatFiles,
+  runTasksInSerial,
+  logger,
+} from '@nx/devkit';
+import { addPlugin } from '@nx/devkit/src/utils/add-plugin';
+import { InitGeneratorSchema } from './schema';
+import { createNodesV2 } from '../../plugins/plugin';
+import { nxVersion } from '../../utils/versions';
+
+export function updateDependencies(tree: Tree, schema: InitGeneratorSchema) {
+  return addDependenciesToPackageJson(
+    tree,
+    {},
+    {
+      '@nx/docker': nxVersion,
+    },
+    undefined,
+    schema.keepExistingVersions
+  );
+}
+
+export function initGenerator(tree: Tree, schema: InitGeneratorSchema) {
```

Comment:
This can be removed and the Internal one can be renamed to be the one and only generator function.